activesupport/lib/active_support/evented_file_update_checker.rb

```diff
@@ -107,13 +107,20 @@ def execute_if_updated
 
     private
       def boot!
+        normalize_dirs!
         Listen.to(*@dtw, &method(:changed)).start
       end
 
       def shutdown!
         Listen.stop
       end
 
+      def normalize_dirs!
+        @dirs.transform_keys! do |dir|
+          dir.exist? ? dir.realpath : dir
+        end
```

Comment:
Does the thing about too many sys calls in Action View apply here? Should we try to spare some SysCalls with something like this?

```ruby
@dirs.transform_keys! do |dir|
  dir.realpath
rescue
  dir
end
```

Reply:
Good question. I don't think that's necessary. `@dirs` are only the top-level dirs being watched, and we probably only run this once on boot (unless a dir is missing and then is created, which should be very rare).

```
> ObjectSpace.each_object(ActiveSupport::EventedFileUpdateChecker).sum{ |x| x.instance_variable_get(:@dirs).size }
# => 28
```

Reply:
Right, so that would be `28 * 2 = 56 syscalls`. Probably not worth it. 👍